src/resource.c

```diff
@@ -735,6 +735,15 @@ static const uint16_t cache_ignore_options[] = { COAP_OPTION_ETAG };
 
   assert( session );
 
+  /* Check if there is already maximum number or subscribers present */
+#if (COAP_RESOURCE_MAX_SUBSCRIBER > 0)
+  uint8_t subscriber_count = 0;
```

Comment:
It is possible that you may get an unexpected wrap around here (if, say, COAP_RESOURCE_MAX_SUBSCRIBER was 256 or larger).  I would increase the size of this variable.